tests/test_tasks.py

```diff
@@ -1,118 +1,137 @@
 import asyncio
-import logging
-import time
-from asyncio import AbstractEventLoop
-from typing import TYPE_CHECKING, Any, Dict, List, NoReturn, Tuple
+from typing import NoReturn
 
 import pytest
-from pytest import LogCaptureFixture
-
-if TYPE_CHECKING:
-    from pytest.capture import CaptureFixture
-else:
-    CaptureFixture = Any
+from pytest_mock import MockerFixture
 
 from fastapi_restful.tasks import repeat_every
 
-logging.basicConfig(level=logging.INFO)
 
+@pytest.mark.asyncio
+async def test_repeat_every_with_sync_function_and_max_repetitions(mocker: MockerFixture) -> None:

```

Comment:
when re-thinking about it a nice way to reimplment these test can be in classes- in which each class has it's on increase_counter fixture (when the fixture define the function so you can still use arguments defined in setup_method - that yields or returnes said function) and a test- sadly it does not prevent code duplication but it can make the test more readable and has less logic in it (just the asserts)- also the setup_mehod can be inherited from an TestAbstract class with it's on setup_method- so it will prevent code duplication.

Reply:
Not sure if that's what you meant, but I did something.